lib/AST/DeclContext.cpp

```diff
@@ -287,6 +287,12 @@ DeclContext *DeclContext::getModuleScopeContext() const {
   }
 }
 
+void DeclContext::getSeparatelyImportedOverlays(
+    ModuleDecl *declaring, SmallVectorImpl<ModuleDecl *> &overlays) const {
+  if (auto SF = getParentSourceFile())
+    SF->getSeparatelyImportedOverlays(declaring, overlays);
```

Comment:
Do we want to pick up overlays introduced by `@_exported` cross-imports in other source files?

Reply:
I think we do. Consider the case where you import, say, UIKit and MyModule, and there's a cross-import available for MyModule and Foundation. Shouldn't we import it? You would certainly expect to be able to call those APIs.

Reply:
@brentdax Sorry, what I mean is that if you have for example a cross-import for module `X` and module `Y`, and in a file `b.swift` you have:

```swift
@_exported import X
@_exported import Y
```

Then shouldn't another file in the same module as `b.swift`, say `a.swift`, also pickup the cross-import overlay? As far as I can tell, the current implementation will only check for cross-imports from `a.swift`'s imports.

Reply:
This is a very, very good catch. I don't want to do even more work without merging, so I'll fix this in a follow-up PR.

Reply:
I've filed a radar on myself for this and added a comment describing the issue.